consensus/src/epoch_manager.rs

```diff
@@ -871,6 +875,8 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
             pipeline_backpressure_config,
             chain_health_backoff_config,
             self.quorum_store_enabled,
+            self.sys_txn_pool_client.clone(),
+            onchain_consensus_config.should_propose_system_txns(),
```

Comment:
don't need it to be a separate field? just have accessor function to read it?

Reply:
could you elaborate?
do you mean `ProposalGenerator` should keep the entire `OnChainConsensusConfig`?